### PR TITLE
add adjoint for var and std

### PR DESCRIPTION
**Pull request by chengchingwen**

### CODE REVIEW

File: src/lib/array.jl [hunk 1]
```diff
@@ -201,6 +201,21 @@ end
 _backmean(xs, Δ, ::Colon) = zero(xs) .+ Δ ./ length(xs)
 _backmean(xs, Δ, dims) = zero(xs) .+ Δ ./ mapreduce(i -> size(xs,i),*,dims)
 
+@adjoint function Statistics.var(xs::AbstractArray; corrected::Bool=true, mean=nothing, dims=:)
+    return Statistics.var(xs; corrected=corrected, mean=mean, dims=dims), Δ -> _backvar(xs, Δ, corrected, mean, dims)
```
Comment (MikeInnes) [inline src/lib/array.jl]: I'm curious why you haven't done `mean = mean(xs)` here. It seems like that'd save some duplicate work.

Comment (chengchingwen) [reply src/lib/array.jl]: You are right! I forgot I can do that

Comment (chengchingwen) [reply src/lib/array.jl]: updated

**Comment by chengchingwen** [comment]
Bump

**Comment by MikeInnes** [comment]
Thanks!

**Pull request merged**